actionview/lib/action_view/dependency_tracker.rb

```diff
@@ -142,8 +142,20 @@ def add_static_dependency(dependencies, dependency)
           end
         end
 
+        def resolve_directories(wildcard_dependencies)
+          wildcard_dependencies.map do |dependency|
+            Dir[ "digestor/#{dependency}" ].map do |file|
```

Comment:
So this is where the hack is. I can't figure out how to reliably find a way to prefix the dependency path with the correct directory nesting.

Some ideas I've been throwing around:

- Extend Action View's resolvers to support wildcard lookups (they have the most info about paths and templates and such).
- Inject a shared prefix path (perhaps as a threaded mattr_accessor)

Either way, the problem is that the prefix is fixed here: https://github.com/rails/rails/blob/094571b46dbc00a2db0bfcc96c38d984d2b675f3/actionview/test/template/digestor_test.rb#L37

Which probably doesn't work the same way as it does outside of tests either.

@rafaelfranca would you happen to have any ideas? :smile: